guava-tests/test/com/google/common/io/BaseEncodingTest.java

```diff
@@ -51,6 +54,34 @@ public void testSeparatorsExplicitly() {
     testEncodes(base32().withSeparator("*", 4), "foobar", "MZXW*6YTB*OI==*====");
   }
 
+  public void testBase64UrlWithPlus() {
+    try {
+      base64Url().decode("+zzz");
+      fail("Expected IllegalArgumentException");
+    } catch (IllegalArgumentException expected) {
+    }
+  }
+
+  public void testBase64UrlWithSlash() {
+    try {
+      base64Url().decode("/zzz");
+      fail("Expected IllegalArgumentException");
+    } catch (IllegalArgumentException expected) {
+    }
+  }
+
+  public void testBase64UrlWithUnderline() {
```

Comment:
Can you use the testDecodes() helper method for these?

Reply:
testDecodes() is use UTF-8 set to encoding, this may limit the scope of BASE64 decoding and make it failed.

I added a new test function names testDecodesByBytes() to compare decode bytes directly.